controllers/user.js

```diff
@@ -992,11 +995,11 @@ async function getRemovalConfirmationPage(req, res) {
 async function getRemovalUpdateConfirmationPage(req, res) {
   const user = req.user;
 
-  if (!user.kid && !req.query.show) {
+  if (!user.kid && !FormUtils.canShowViaParams(req.query?.show)) {
     return res.redirect("/user/remove-enroll");
   }
 
-  if (user.kid && !req.query.show) {
+  if (user.kid && !FormUtils.canShowViaParams(req.query?.show)) {
```

Comment:
Again, not sure it's "better", or even more readable.
```js
  if (!FormUtils.canShowViaParams(req.query?.show)) {
    if (!user.kid) {
      return res.redirect("/user/remove-enroll");
    }
    return res.redirect("/user/remove-data");
  }
```

Reply:
Going to leave as is unless there's any strong objection.